src/components/10-atoms/checkbox/index.js

```diff
@@ -0,0 +1,174 @@
+      onChange: { type: Function },
+    };
+  }
+
+  static get styles() {
+    return styles;
+  }
+
+  constructor() {
+    super();
+    // initialize model state
+    this.state = {
+      isControlled: false,
+      timer: null,
+      checked: false,
+      native: false,
+    };
+    // initialize properties
+    this.type = 'checkbox';
+    this.value = '';
+    this.name = '';
+    this.label = '';
+    this.checked = false;
+    this.disabled = false;
+    this.error = '';
+    this.isReact = false;
+    this.onChange = () => {};
+  }
+
+  // custom setter
+  set checked(value) {
+    const {
+      state: { isControlled, checked },
+    } = this;
+    // first incoming value indicates controlledness?
+    if (!isControlled && value !== undefined) {
+      // yes, remember in model state
+      this.state.isControlled = true;
+    }
+    const oldValue = checked;
+    // remember value in model state
+    this.state.checked = value;
+    this.state.native = value;
+    // request re-render (custom setters need to do this themselves!)
+    this.requestUpdate('checked', oldValue);
+  }
+
+  // custom getter
+  get checked() {
+    const {
+      state: { isControlled, checked, native },
+    } = this;
+    return isControlled ? checked : native;
+  }
+
+  handleChange(event) {
+    // initialize
+    const {
+      state: { native },
+      onChange,
+    } = this;
+    // *schedule* a UI update from model state in the near future
+    // (if no changed prop value is seen, triggering re-rendering,
+    // that scheduled update actually happens!)
+    this.state.timer = setTimeout(() => this.updated(), 0);
+    // toggle *uncontrolled* model state to reflect checkbox behaviour
+    this.state.native = !native;
+    // invoke event callback
+    onChange(event);
+  }
+
+  render() {
+    // extract props and state
+    const {
+      type,
+      value,
+      name,
+      label = '',
+      checked,
+      disabled,
+      error = '',
+      id,
+      isReact,
+      state: { isControlled, timer },
+    } = this;
+    // now that we have the 'isReact' prop, determine if this
+    // component is a 'controlled input' in the *React* sense
+    const _isControlled = isControlled && isReact;
+    this.state.isControlled = _isControlled;
+    if (_isControlled) {
+      // cancel any scheduled UI update, since there is a real,
+      // changed prop value somewhere (likely 'checked')
+      clearTimeout(timer);
+    }
+    return html`
+      <label class="a-checkbox__wrapper">
+        <input
+          id="${id}"
+          class="a-checkbox__input"
+          type="${type}"
+          name="${name}"
+          ?checked="${checked}"
+          value="${value}"
+          ?disabled="${disabled}"
+          ?error="${!!error}"
+          @change=${this.handleChange}
+        />
+        <div class="a-checkbox__icon"></div>
+        ${label &&
+          html`
+            <div class="a-checkbox__content">${unsafeHTML(label)}</div>
```

Comment:
`<div>` is not allowed inside a label, only phrasing content. Please use one out of: `a abbr area (if it is a descendant of a map element) audio b bdi bdo br button canvas cite code data datalist del dfn em embed i iframe img input ins kbd keygen label map mark math meter noscript object output progress q ruby s samp script select small span strong sub sup svg textarea time u var video wbr text`

REF: 
- https://html.spec.whatwg.org/multipage/forms.html#the-label-element
- https://html.spec.whatwg.org/multipage/dom.html#phrasing-content-2

Reply:
Good catch - I initially copied from v1 HTML... Changed everywhere to span with display:block.